dags/backups.py

```diff
@@ -197,6 +201,20 @@ class BackupConfig(dagster.Config):
     )
 
 
+def get_most_recent_status(statuses: list[BackupStatus]) -> Optional[BackupStatus]:
+    """
+    Since we can retry backups and we only can identify them by their name (or path to S3),
+    in case we retry several times, we can have the same backup failed in one node and
+    succeeded in another one.
+
+    This function will raise an error only if the most recent one didn't succeed
+    """
+    statuses = [status for status in statuses if status is not None]
+    if statuses:
+        statuses.sort(key=lambda x: x.event_time_microseconds, reverse=True)
+        return statuses[0]
```

Comment:
if you only need min/max you could use:
return max(statuses, key=lambda x: x.event_time_microseconds)